platform/api-server/api/controllers/platform-api-controller.js

```diff
@@ -99,3 +103,244 @@ exports.updatePrivacyPolicy = async function(req, res) {
     });
     res.send();
 };
+
+/** Launch an application on a suitable gateway.
+ *
+ * @param req Request information.
+ * @param res Response object.
+ */
+exports.deployApplication = async function(req, res) {
```

Comment:
similar to execute-app-v2, it would be best to do the tar extraction and scheduling in the app-manager. maybe create a file called scheduler.js in the app-manager/ directory. When building the resiliency features, I had built that workflow. You can refer to it here:
https://github.com/nabeeln7/on-the-edge/tree/nabeel/resiliency-app-restart